plugins/module_utils/mso.py

```diff
@@ -776,32 +776,38 @@ def get_obj(self, path, api_version="v1", **kwargs):
             self.fail_json(msg="More than one object matches unique filter: {0}".format(kwargs))
         return objs[0]
 
-    def lookup_schema(self, schema):
+    def lookup_schema(self, schema, ignore_404=False):
```

Comment:
Few questions ->
1. Do we need to flag ignore_404 as True for all the modules using lookup_schema, lookup_site etc. when the state is absent?
2. If the answer is yes to Q1 can we just modify the functions lookup_schema, lookup_site and etc. without making any change to all the other modules?
Example -
```
def lookup_schema(self, schema):
"""Look up schema and return its id"""
        if schema is None:
            return schema

        schema_summary = self.query_objs("schemas/list-identity", key="schemas", displayName=schema)
        current_state = self.params.get("state")
        if not schema_summary and current_state is not "absent":
            self.fail_json(msg="Provided schema '{0}' does not exist.".format(schema))
        schema_id = schema_summary[0].get("id")
        if not schema_id and current_state is not "absent":
            self.fail_json(msg="Schema lookup failed for schema '{0}': '{1}'".format(schema, schema_id))
        return schema_id
```
3. If we agree on using the flag ignore_404 we could change ignore_404 -> ignore_not_found_error? Let's consult with the others before changing.

Reply:
Sure @shrsr